main/src/toolhive-manager.ts

```diff
@@ -104,65 +105,81 @@ async function findFreePort(
 }
 
 export async function startToolhive(tray?: Tray): Promise<void> {
-  if (!existsSync(binPath)) {
-    log.error(`ToolHive binary not found at: ${binPath}`)
-    return
-  }
+  Sentry.withScope<Promise<void>>(async (scope) => {
+    if (!existsSync(binPath)) {
+      log.error(`ToolHive binary not found at: ${binPath}`)
+      return
+    }
 
-  log.info(`APP USER DATA: ${app.getPath('userData')}`)
-  toolhivePort = await findFreePort(50000, 50100)
-  toolhiveMcpPort = await findFreePort()
-  log.info(
-    `Starting ToolHive from: ${binPath} on port ${toolhivePort}, MCP on port ${toolhiveMcpPort}`
-  )
+    toolhivePort = await findFreePort(50000, 50100)
+    toolhiveMcpPort = await findFreePort()
+    log.info(
+      `Starting ToolHive from: ${binPath} on port ${toolhivePort}, MCP on port ${toolhiveMcpPort}`
+    )
+
+    toolhiveProcess = spawn(
+      binPath,
+      [
+        'serve',
+        '--openapi',
+        '--experimental-mcp',
+        '--experimental-mcp-host=127.0.0.1',
+        `--experimental-mcp-port=${toolhiveMcpPort}`,
+        '--host=127.0.0.1',
+        `--port=${toolhivePort}`,
+      ],
+      {
+        stdio: ['ignore', 'ignore', 'pipe'],
+        detached: false,
+      }
+    )
+
+    log.info(`[startToolhive] Process spawned with PID: ${toolhiveProcess.pid}`)
 
-  toolhiveProcess = spawn(
-    binPath,
-    [
-      'serve',
-      '--openapi',
-      '--experimental-mcp',
-      '--experimental-mcp-host=127.0.0.1',
-      `--experimental-mcp-port=${toolhiveMcpPort}`,
-      '--host=127.0.0.1',
-      `--port=${toolhivePort}`,
-    ],
-    {
-      stdio: ['ignore', 'ignore', 'pipe'],
-      detached: false,
+    scope.addBreadcrumb({
+      category: 'debug',
+      message: `Starting ToolHive from: ${binPath} on port ${toolhivePort}, MCP on port ${toolhiveMcpPort}, PID: ${toolhiveProcess.pid}`,
+    })
+
+    if (tray) {
+      updateTrayStatus(tray, !!toolhiveProcess)
     }
-  )
 
-  log.info(`[startToolhive] Process spawned with PID: ${toolhiveProcess.pid}`)
+    // Capture and log stderr
+    if (toolhiveProcess.stderr) {
+      log.info(`[ToolHive] Capturing stderr enabled`)
+      toolhiveProcess.stderr.on('data', (data) => {
+        const output = data.toString().trim()
+        if (output) {
+          scope.addBreadcrumb({
```

Comment:
I would add the `log.error(`[ToolHive stderr] ${output}`)` as well in order to have the correct flow, and also everything visible and not scrubbed